src/frontend/js/components/DashboardItem/DashboardItemCourseEnrolling.tsx

```diff
@@ -0,0 +1,301 @@
+          ),
+        }))
+        .filter(
+          (data) => data.selected || data.courseRun.state.priority <= Priority.FUTURE_NOT_YET_OPEN,
+        ),
+    [course, enrollments],
+  );
+
+  // Filter by priority
+  return (
+    <div className="dashboard-item__course-enrolling__runs">
+      {error && <Banner message={error} type={BannerType.ERROR} />}
+      {datas.length === 0 && (
+        <div className="dashboard-item__course-enrolling__no-runs">
+          <Icon name="icon-warning" size="small" />
+          <FormattedMessage {...messages.noCourseRunAvailable} />
+        </div>
+      )}
+      {datas.map((data) => (
+        <DashboardItemCourseEnrollingRun
+          key={data.courseRun.id}
+          courseRun={data.courseRun}
+          selected={data.selected}
+          enroll={() => enroll(data.courseRun)}
+        />
+      ))}
+      {isLoading && (
+        <div
+          className="dashboard-item__course-enrolling__loading"
+          data-testid="dashboard-item__course-enrolling__loading"
+        >
+          <Spinner aria-labelledby="enrolling-loading" size="large">
+            <span id="enrolling-loading">
+              <FormattedMessage {...messages.courseRunsLoading} />
+            </span>
+          </Spinner>
+        </div>
+      )}
+    </div>
+  );
+};
+
+const DashboardItemCourseEnrollingRun = ({
+  courseRun,
+  selected,
+  enroll,
+}: {
+  courseRun: CourseRun;
+  selected: boolean;
+  enroll: () => void;
+}) => {
+  const formatDate = useDateFormat();
+  const isOpenedForEnrollment = useMemo(
+    () => courseRun.state.priority < Priority.FUTURE_NOT_YET_OPEN,
+    [courseRun],
+  );
+  return (
+    <div
+      className="dashboard-item__course-enrolling__run"
+      data-testid={'dashboard-item__course-enrolling__run__' + courseRun.id}
+    >
+      <div>
+        <div>
+          <FormattedMessage
+            {...messages.runPeriod}
+            values={{
+              startDate: formatDate(courseRun.start, DATETIME_FORMAT),
+              endDate: formatDate(courseRun.end, DATETIME_FORMAT),
+            }}
+          />
+        </div>
+        {!isOpenedForEnrollment && (
+          <div className="dashboard-item__course-enrolling__run__not-opened">
+            <FormattedMessage
+              {...messages.enrollmentNotYetOpened}
+              values={{ enrollment_start: formatDate(courseRun.enrollment_start) }}
+            />
+          </div>
+        )}
+      </div>
+      <div>
+        {selected && (
+          <div className="dashboard-item__course-enrolling__run__enrolled">
+            <FormattedMessage {...messages.enrolled} />
+            <Icon name="icon-check" size="small" />
+          </div>
+        )}
+        {!selected && (
+          <Button disabled={!isOpenedForEnrollment} color="outline-primary" onClick={enroll}>
+            <FormattedMessage {...messages.enrollRun} />
+          </Button>
+        )}
```

Comment:
I think a ternary operator is better for maintainability in this case.
```suggestion
        {selected ? (
          <div className="dashboard-item__course-enrolling__run__enrolled">
            <FormattedMessage {...messages.enrolled} />
            <Icon name="icon-check" size="small" />
          </div>
        ) : (
          <Button disabled={!isOpenedForEnrollment} color="outline-primary" onClick={enroll}>
            <FormattedMessage {...messages.enrollRun} />
          </Button>
        )}
```